app/Http/Controllers/CampaignsController.php

```diff
@@ -43,22 +43,29 @@ public function __construct(Registrar $registrar, CampaignService $campaignServi
     public function index()
     {
         $ids = $this->campaignService->getCampaignIdsFromSignups();
+
         $campaigns = $this->campaignService->findAll($ids);
-        $campaigns = $this->campaignService->groupByCause($campaigns);
+
+        $campaigns = $campaigns->map(function($campaign, $key) {
+            if ($campaign) {
+                $campaign = $this->campaignService->getCampaignPostStatusCounts($campaign);
```

Comment:
I think you should be able to pass in the array of campaigns to that scope you defined on the model:

```php
$ids = array_pluck($campaign, 'id');
$campaigns = Signup::campaign($ids)
    ->includePostStatusCounts()
    ->groupBy('campaign_id')
    ->get();

$campaigns = $campaigns->map(function($campaign) {
    $campaign['accepted_count'] = $campaign->sum('accepted_count');
    $campaign['pending_count'] = $campaign->sum('pending_count');
    $campaign['rejected_count'] = $campaign->sum('rejected_count');

    return $campaign;
});
```